File: Cpp/682_Baseball_Game.cpp
class Solution {
public:
    int calPoints(vector<string>& ops) {
        vector<int> s;
        
        for(auto& ch : ops){
            if(ch == "+"){
                s.push_back(s.back() + s[s.size()-2]);
            } else if(ch == "C"){
                s.pop_back();
            } else if(ch == "D"){
                s.push_back(s.back()*2);
            } else {
                s.push_back(stoi(ch));
            }
        }
        
        return accumulate(s.begin(), s.end(), 0);
    }
};

File: Kotlin/682_Baseball_Game.kt
class Solution {
    fun calPoints(ops: Array<String>): Int {
        val s = Stack<Int>()
        
        for(operation in ops) {
            when(operation) {
                "+" -> s.push(s[s.size - 1] + s[s.size - 2])
                "D" -> s.push(s.peek() * 2)
                "C" -> s.pop()
                else -> s.push(operation.toInt())
            }
        }
        
        return s.sum()
    }
}